options/posix/generic/services.cpp

```diff
@@ -83,39 +84,43 @@ static int lookup_serv_file(struct service_buf *buf, const char *name,
 }
 
 int lookup_serv(struct service_buf *buf, const char *name, int proto,
-		int socktype) {
+		int socktype, int flags) {
 	switch(socktype) {
 		case SOCK_STREAM:
 			if (!proto)
 				proto = IPPROTO_TCP;
 			else if (proto != IPPROTO_TCP)
-				return EAI_SERVICE;
+				return -EAI_SERVICE;
```

Comment:
Why the minus?

Reply:
A positive return value indicates success, so I return a negative one, and then make it positive again in ``getaddrinfo()``

Reply:
Okay, maybe add a comment to state that it works like this (?).

Other than that, the PR LGTM.